### PR TITLE
.

### PR DESCRIPTION
**Pull request by erbsenzaehler**
It seems that icc used our fallback version of popcount. Now use intrinsics.

icc version 16.0.2 (gcc version 5.3.0 compatibility)
bmi2 compile
uname -r 4.5.1-1-ARCH

20xbench gives a nice speedup
./stockfish-icc-master 2161515 +- 34462	
./stockfish-icc-sse42	 2260857 +- 50349

### CODE REVIEW

File: src/types.h [hunk 1]
```diff
@@ -64,7 +64,7 @@
 #  define IS_64BIT
 #endif
 
-#if defined(USE_POPCNT) && defined(__INTEL_COMPILER) && defined(_MSC_VER)
+#if defined(USE_POPCNT) && defined(__INTEL_COMPILER) || defined(_MSC_VER)
```
Comment (snicolet) [inline src/types.h]: You should probably use parenthesis here to clarify the order of evaluation.

**Comment by lucasart** [comment]
Popcount does not imply sse4.2. Only the reverse is true though. In other words , there are machines with only sse3, that have popcount (some AMD, dont remember which). Please dont break that, and use sse3+popcount separately, instead of sse4.2

**Comment by erbsenzaehler** [comment]
I added the changes. Using sse3 over sse4.2  gives a small decrease in nps but icc is still behing clang+gcc speedwise.

I find the intel compiler manual very confusing in this regard...

**Comment by mcostalba** [comment]
Merged.

**Pull request closed without merging**